flycheck.el

```diff
@@ -8622,7 +8622,7 @@ for more information about the custom directories."
   "Whether there is a valid eslint config for the current buffer."
   (let* ((executable (flycheck-find-checker-executable 'javascript-eslint))
          (exitcode (and executable (call-process executable nil nil nil
-                                                 "--print-config" "."))))
+                                                 "--print-config" "./__placeholder__.js"))))
```

Comment:
Wait, what?  Passing a directory is an error, but passing a phony filename isn't?  What is the reasoning behind this change in ESLint?

Reply:
It probably matches against some pattern, it check which config is valid for that pattern to actual code. A directory is not code.

Reply:
I think you can and should use `buffer-file-name` if available.  What about `(or buffer-file-name "index.js")` instead?

Reply:
I see.  But then shouldn't we pass the filename of the current buffer instead?  Since it could return a different configuration?

Reply:
Fixed!